scrapy/core/downloader/__init__.py

```diff
@@ -99,9 +100,14 @@ def needs_backout(self):
     def _get_slot(self, request, spider):
         key = self._get_slot_key(request, spider)
         if key not in self.slots:
-            conc = self.ip_concurrency if self.ip_concurrency else self.domain_concurrency
+            conc = self.per_slot_settings.get(key, {}).get(
+                'concurrency', self.ip_concurrency if self.ip_concurrency else self.domain_concurrency
+            )
             conc, delay = _get_concurrency_delay(conc, spider, self.settings)
-            self.slots[key] = Slot(conc, delay, self.randomize_delay)
+            delay = self.per_slot_settings.get(key, {}).get('delay', delay)
+            randomize_delay = self.per_slot_settings.get(key, {}).get('randomize_delay', self.randomize_delay)
```

Comment:
A nitpick: what do you think about adding this variable:
```py
slot_settings = self.per_slot_settings.get(key, {})
```
It seems it's used a few times here.

Reply:
Done